src/sentry/static/sentry/app/components/confirm.tsx

```diff
@@ -14,268 +13,243 @@ type MessageRenderProps = {
+    disableConfirmButton: !!this.props.disableConfirmButton,
   };
 
+  confirming: boolean = false;
+
   render() {
     const {
-      disabled,
-      message,
-      renderMessage,
+      Header,
+      Body,
+      Footer,
       priority,
+      renderMessage,
+      message,
       confirmText,
       cancelText,
-      children,
       header,
+      onConfirm,
+      onCancel,
+      disableConfirmButton,
+      closeModal,
     } = this.props;
 
-    let confirmMessage: React.ReactNode;
-
-    if (typeof renderMessage === 'function') {
-      confirmMessage = renderMessage({
-        confirm: this.handleConfirm,
-        close: this.handleToggle,
-      });
-    } else {
-      confirmMessage = React.isValidElement(message) ? (
+    const handleClose = () => {
+      onCancel?.();
+      this.setState({disableConfirmButton: disableConfirmButton ?? false});
+
+      // always reset `confirming` when modal visibility changes
+      this.confirming = false;
+      closeModal();
+    };
+
+    const handleConfirm = () => {
+      // `confirming` is used to make sure `onConfirm` is only called once
+      if (!this.confirming) {
+        onConfirm();
+      }
+
+      this.setState({disableConfirmButton: true});
+      this.confirming = true;
+      closeModal();
+    };
+
+    const confirmMessage =
+      typeof renderMessage === 'function' ? (
+        renderMessage({
+          confirm: handleConfirm,
+          close: handleClose,
+          disableConfirmButton: (state: boolean) =>
+            this.setState({disableConfirmButton: state}),
+        })
+      ) : React.isValidElement(message) ? (
         message
       ) : (
         <p>
           <strong>{message}</strong>
         </p>
       );
-    }
 
     return (
       <React.Fragment>
-        {typeof children === 'function'
-          ? children({
-              close: this.closeModal,
-              open: this.openModal,
-            })
-          : React.isValidElement(children) &&
-            React.cloneElement(children, {
-              disabled,
-              onClick: this.handleToggle,
-            })}
-        <Modal show={this.state.isModalOpen} animation={false} onHide={this.handleToggle}>
-          {header && <Modal.Header>{header}</Modal.Header>}
-          <Modal.Body>{confirmMessage}</Modal.Body>
-          <Modal.Footer>
-            <Button style={{marginRight: 10}} onClick={this.handleToggle}>
-              {cancelText}
-            </Button>
-            <Button
-              data-test-id="confirm-button"
-              disabled={this.state.disableConfirmButton}
-              priority={priority}
-              onClick={this.handleConfirm}
-              autoFocus
-            >
-              {confirmText}
-            </Button>
-          </Modal.Footer>
-        </Modal>
+        {header && <Header>{header}</Header>}
+        <Body>{confirmMessage}</Body>
+        <Footer>
+          <Button style={{marginRight: 10}} onClick={handleClose}>
```

Comment:
can we style this button using emotion and the space utility?

Reply:
haha I saw this and didn't change it and knew i would get this comment

Reply:
Done! Using `ButtonBar`